### PR TITLE
Use a builder instead of child in AnimatedBookWidget

### PR DESCRIPTION
**Pull request by ecarlosdev**
## Description

Use a builder instead of child in AnimatedBookWidget #18

## Checklist

- [x] Tested on iOS
- [ ] Tested on Android
- [ ] Tested on small screens
- [ ] Unit tests

### CODE REVIEW

**Review by br-programmer — APPROVED** [review]

**Review by diegoveloper — CHANGES_REQUESTED** [review]
I don't think we should let the user to implement the animation by them self.
The idea of the package is to simplify the integration/code, so that option must be optional, not required.

I was thinking in something like this: 
```dart
const AnimatedBookWidget({
    required this.cover,
    required this.size,
    this.content,
    this.contentBuilder,
    this.contentChild,
    ... 
```
We could use either `content` or `contentBuilder` but not both at the same time.
If we use content, the animation is included automatically (like the widget is right now, not on this PR).
If we use contentBuilder, then we can use your way, also if contentWidget is defined, we can use that child in the builder, here: 

```
contentBuilder: (context, bookAnimation, child) {
``` 

we should specify the usage in the documentation of the method.

**Comment by ecarlosdev** [comment]
What do you think about this? @diegoveloper 

``` dart
AnimatedBookWidget(content: ColoredBox(color: color))
```

``` dart
AnimatedBookWidget.custom(
    contentBuilder: (context, bookAnimation, child) {
        return SizedBox.expand(
            child: Transform(
                alignment: Alignment.center,
                transform: Matrix4.identity()
                    ..scale(bookAnimation.value),
                child: child,
            ),
        );
    },
    contentChild: ColoredBox(color: color),
)
```

**Comment by diegoveloper** [comment]
> What do you think about this? @diegoveloper
> 
> ```dart
> AnimatedBookWidget(content: ColoredBox(color: color))
> ```
> 
> ```dart
> AnimatedBookWidget.custom(
>     contentBuilder: (context, bookAnimation, child) {
>         return SizedBox.expand(
>             child: Transform(
>                 alignment: Alignment.center,
>                 transform: Matrix4.identity()
>                     ..scale(bookAnimation.value),
>                 child: child,
>             ),
>         );
>     },
>     contentChild: ColoredBox(color: color),
> )
> ```

It looks good, but instead of custom use .builder

File: lib/src/widgets/animated_content_widget.dart [hunk 1]
```diff
@@ -1,29 +1,95 @@
...
+      animation: bookAnimation,
+      builder: (context, child) =>
+          delegate.build(context, bookAnimation, child),
+      child: delegate.contentChild,
+    );
+  }
+}
+
+///
+abstract class AnimatedContentDelegate {
```
Comment (diegoveloper) [inline lib/src/widgets/animated_content_widget.dart]: can you move this class and the implementation classes into a new file? 'animated_content_delegate.dart' file

Comment (ecarlosdev) [reply lib/src/widgets/animated_content_widget.dart]: Sure. In which directory should I put it?

Comment (diegoveloper) [reply lib/src/widgets/animated_content_widget.dart]: same level as "animated_content_widget.dart"

Comment (ecarlosqm) [reply lib/src/widgets/animated_content_widget.dart]: Done

Comment (diegoveloper) [reply lib/src/widgets/animated_content_widget.dart]: wait, are you using two github accounts? xd

Comment (ecarlosdev) [reply lib/src/widgets/animated_content_widget.dart]: haha, Yes I have two accounts. I made a mistake

**Review by diegoveloper — APPROVED** [review]

**Review by br-programmer — APPROVED** [review]
LGTM!

**Pull request merged**